### ISSUE TITLE
Other correlation time estimators

### ISSUE THREAD

**Issue by rmcgibbo** [rmcgibbo/pyhmc#3]
1. [Sokal's notes (page 16)](http://www.stat.unc.edu/faculty/cji/Sokal.pdf), citing [an earlier paper](http://link.springer.com/article/10.1007/BF01022990#page-1) use an procedure for choosing the window for summing the empirical ACF where the length of the window, `M` is chosen to be the smallest value such that `M` is at least `c` times the estimated autocorrelation time, where `c` is something like 4, 6, or 10.
2. [Jenke](http://www.physik.uni-leipzig.de/~janke/Paper/nic10_423_2002.pdf) also uses the same scheme, with c=6.
3. Evertz doesn't seem to like this procedure [(page 59)](https://itp.tugraz.at/~evertz/Pubs/2003_AdvPhys52_1_cm9797221v3_LoopReview.pdf)
4. LaplacesDemon's in R uses the Geyer IPS estimator.
5. Geyer's ICS estimator.

**Comment by jchodera** [comment]
In case you haven't seen [Geyer's chapter from the Handbook of Markov chain Monte Carlo](http://www.mcmchandbook.net/HandbookChapter1.pdf), it's has some nice discussion of this as well in Section 1.10 "Variance Estimation".

I'd love to see the *initial convex sequence* (ICS) estimator be implemented as well (discussed in Section 1.10.2).   The canonical reference for this and related methods seems to be:
> Geyer, C. J. 1992. Practical Markov chain Monte Carlo (with discussion). Statistical Science, 7:473–511.
PDF available [here](http://projecteuclid.org/euclid.ss/1177011137).

**Comment by rmcgibbo** [comment]
Yes, I've read the paper and cited it in the [IPS docs](https://pythonhosted.org/pyhmc/generated/pyhmc.integrated_autocorr3.html#pyhmc.integrated_autocorr3). Adding the ICS is a good idea too.

**Comment by rmcgibbo** [comment]
Also there's another ACF from http://www.sciencedirect.com/science/article/pii/0370269389915633 that I just saw, where you truncate the summation after a fixed number of terms (e.g. 20), and then add in a remainder based on the ratio of the final two values included in the sum, which looks like an analytical correction based on the assumption that the terms are decaying exponentially.
